suite-common/wallet-utils/src/__tests__/accountUtils.test.ts

```diff
@@ -337,4 +340,113 @@ describe('account utils', () => {
+
+    type RunFilterReceiveAccountsTestParams = {
+        isDebug?: boolean;
+        receiveNetwork?: string;
+        deviceState?: `${string}@${string}:${number}`;
+    };
+
+    const runFilterReceiveAccouns = ({
+        isDebug = true,
+        receiveNetwork = 'eth',
+        deviceState = '1stTestnetAddress@device_id:0',
+    }: RunFilterReceiveAccountsTestParams) => {
+        const device = getSuiteDevice({
+            unavailableCapabilities: {
+                dash: 'no-support',
+            },
+            state: deviceState,
+        });
+        const unavailableCapabilities = device?.unavailableCapabilities ?? {};
+
+        const receiveNetworks = networksCollection.filter(
+            (n: Network) =>
+                n.symbol === receiveNetwork &&
+                !unavailableCapabilities[n.symbol] &&
+                ((n.isDebugOnlyNetwork && isDebug) || !n.isDebugOnlyNetwork),
+        );
+
+        return filterReceiveAccounts({
+            accounts,
+            deviceState: device.state,
+            receiveNetwork,
+            isDebug,
+            receiveNetworks,
+        });
+    };
+
+    it('filterReceiveAccounts: isDebug', () => {
+        const filteredAccounts = [
+            getWalletAccount({ symbol: 'eth', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'normal' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'ledger' }),
+        ];
+
+        expect(runFilterReceiveAccouns({})).toEqual(filteredAccounts);
+    });
+
+    it('filterReceiveAccounts: isNotDebug', () => {
+        const filteredAccounts = [getWalletAccount({ symbol: 'eth', accountType: 'normal' })];
+
+        expect(runFilterReceiveAccouns({ isDebug: false })).toEqual(filteredAccounts);
+    });
+
+    it('filterReceiveAccounts: isNotSameDevice', () => {
+        expect(runFilterReceiveAccouns({ deviceState: '2ndTestnetAddress@device_id:0' })).toEqual(
+            [],
+        );
+    });
+
+    it('filterReceiveAccounts: isAnotherNetwork, isCoinjoinAccount for btc', () => {
+        const filteredAccounts = [
+            getWalletAccount({ symbol: 'btc', accountType: 'taproot' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'segwit' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'ledger' }),
+        ];
+
+        expect(runFilterReceiveAccouns({ receiveNetwork: 'btc' })).toEqual(filteredAccounts);
+    });
+
+    it('filterReceiveAccounts: isEmptyAccount and isNotVisibleAccount + isEmptyAccount and isNotVisibleAccount but isFirstNormalAccount', () => {
+        const filteredAccounts = [
+            getWalletAccount({ symbol: 'sol', accountType: 'normal', empty: true, visible: false }),
+            getWalletAccount({ symbol: 'sol', accountType: 'ledger' }),
+        ];
+
+        expect(runFilterReceiveAccouns({ receiveNetwork: 'sol' })).toEqual(filteredAccounts);
+    });
```

Comment:
I think we are missing test for:
1. empty accounts array
2. non-existent network in network collection
3. mixed account types - debug and non-debug

Reply:
Not sure what you mean by mixed account types, I have a test for when Debug is on - then we get all the account types - debug and non-debug, when debug is off we get only non-debug, I have a test for that too.